src/singletons/Settings.cpp

```diff
@@ -85,8 +85,7 @@ void ConcurrentSettings::mute(const QString &channelName)
 
 void ConcurrentSettings::unmute(const QString &channelName)
 {
-    for (std::vector<int>::size_type i = 0; i != mutedChannels.raw().size();
-         i++)
+    for (std::size_t i = 0; i != mutedChannels.raw().size(); i++)
```

Comment:
I don't think this change is correct if we check https://en.cppreference.com/w/cpp/container/vector/size and see that the vector indices will be determined by the type of the vector.
Which in this case
https://github.com/Chatterino/chatterino2/blob/50308905f42ce14d0de6c5369a515d58a0e559af/src/singletons/Settings.cpp#L24
is a QString.
So the correct change would be:
```suggestion
    for (std::vector<QString>::size_type i = 0; i != mutedChannels.raw().size(); i++)
```
I think...

Reply:
If you open up the code in an IDE, it's show size() return type as std::size_t.

Reply:
![Screenshot from 2021-08-09 22-38-04](https://user-images.githubusercontent.com/17646853/128771221-f72b775c-80ad-4bad-8fa1-f9a8b33e35e6.png)

Reply:
The `size()` method is defined as follows:
```
      // [23.2.4.2] capacity
      /**  Returns the number of elements in the %vector.  */
      size_type
      size() const _GLIBCXX_NOEXCEPT
      { return size_type(this->_M_impl._M_finish - this->_M_impl._M_start); }
```
(src: `stl_vector.h`)

It actually returns `std::vector<T>::size_type`, not `size_t` directly

Reply:
I'm not really arguing against you.
And probably every compiler agrees with you.

But according the standard this will NOT always return size_t and while your compiler will return size_t doesn't mean it will on every compiler.

(We don't do this anywhere else in the code either so your change is valid.)

Reply:
Even tho you're right, `std::vector<QString>::size_type` is the correct way doing it, it's ugly and never seen it anywhere else.